packages/versioning/src/validate.ts

```diff
@@ -205,6 +210,74 @@ function validateMultiTypeReference(program: Program, source: Type) {
   }
 }
 
+/**
+ * Constructs a map of version to name for the the source.
+ */
+function getVersionedNameMap(
+  program: Program,
+  source: Type
+): Map<Version, string | undefined> | undefined {
+  const allVersions = getAllVersions(program, source);
+  if (allVersions === undefined) return undefined;
+
+  const map: Map<Version, string | undefined> = new Map(allVersions.map((v) => [v, undefined]));
+  const availMap = getAvailabilityMap(program, source);
+  const alwaysAvail = availMap === undefined;
+
+  // Populate the map with any RenamedFrom data, which may have holes.
+  // We will fill these holes in a later pass.
+  const renamedFrom = getRenamedFrom(program, source);
+  if (renamedFrom !== undefined) {
+    for (const rename of renamedFrom) {
+      const version = rename.version;
+      const oldName = rename.oldName;
+      const versionIndex = allVersions.indexOf(version);
+      if (versionIndex !== -1) {
+        map.set(allVersions[versionIndex - 1], oldName);
+      }
+    }
+  }
+  let lastName: string | undefined = undefined;
+  switch (source.kind) {
+    case "ModelProperty":
+      lastName = source.name;
+      break;
+    case "UnionVariant":
+      if (typeof source.name === "string") {
+        lastName = source.name;
+      } else if (typeof source.name === "symbol") {
+        lastName = source.name.description;
```

Comment:
I would just ignore if it is a symbol, point of the symbol is it is anonynous

Reply:
Done